org.eclim.core/java/org/eclim/plugin/core/command/file/FileSaveCommand.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * Warning: you might have to update the <code>project</code> after calling this
+ * command.
+ *
+ * @author Lukas Roth
+ *
+ */
+public class FileSaveCommand extends AbstractCommand
+{
+  private static final Logger logger = Logger.getLogger(FileSaveCommand.class);
+
+  @Override
+  public Object execute(CommandLine commandLine)
+      throws Exception
+  {
+    String relativeFilePath = commandLine.getValue(Options.FILE_OPTION);
+    String projectName = commandLine.getValue(Options.PROJECT_OPTION);
+    Object fileIn = commandLine.getRawValue(Options.CONTENT_OPTION);
+    InputStream fileStream;
+    if (fileIn == null) {
+      fileStream = new ByteArrayInputStream("".getBytes(StandardCharsets.UTF_8));
+    } else if (fileIn instanceof String) {
+      fileStream = new ByteArrayInputStream(
+          ((String) fileIn).getBytes(StandardCharsets.UTF_8));
+    } else if (fileIn instanceof InputStream) {
+      fileStream = (InputStream) fileIn;
+    } else {
+      String message = Services.getMessage("file.save.content.wrong.type");
+      logger.error(message);
+      return new CommandException(message, ErrorType.SYSTEM_ERROR);
+    }
+    return fileSave(relativeFilePath, projectName, fileStream);
+  }
+
+  public Object fileSave(String relativeFilePath, String projectName,
+      InputStream fileContent)
+  {
+    try {
+      PathUtil.checkPathForEscaping(relativeFilePath);
+    } catch (PathUtilException e) {
+      logger.error("File path is not valid", e);
+      return new CommandException(e, ErrorType.CLIENT_ERROR);
+    }
+    String absoluteFilePath;
+    try {
+      absoluteFilePath = PathUtil.getAbsolutePath(projectName, relativeFilePath);
+    } catch (PathUtilException e) {
+      logger.error("Could not get the absolute path", e);
+      return new CommandException(e, ErrorType.CLIENT_ERROR);
+    }
+    try {
+      writeToFileSystem(absoluteFilePath, fileContent);
+      return Services.getMessage("file.save.success", relativeFilePath);
+    } catch (IOException e) {
+      String message = Services.getMessage("file.save.error.io", relativeFilePath);
+      logger.error(message, e);
+      return new CommandException(message, ErrorType.SYSTEM_ERROR);
+    }
+  }
+
+  private void writeToFileSystem(String absoluteFilePath, InputStream fileContent)
+      throws IOException
+  {
+    File file = new File(absoluteFilePath);
+    if (file.exists()) {
+      logger.debug("Overwriting file at " + absoluteFilePath);
+    }
+    file.getParentFile().mkdirs();
+    copyInputStreamToFile(fileContent, file);
+  }
+
+  // copied from
+  // http://stackoverflow.com/questions/43157/easy-way-to-write-contents-of-a-java-inputstream-to-an-outputstream
+  private void copyInputStreamToFile(InputStream in, File file)
+      throws IOException
+  {
+    OutputStream out = new FileOutputStream(file);
+    byte[] buf = new byte[1024];
+    int len;
+    while ((len = in.read(buf)) > 0) {
+      out.write(buf, 0, len);
+    }
```

Comment:
You can remove this code and replace it with `org.eclim.util.IOUtils.copy(in, out)`.